ports/rp2/mpnetworkport.c

```diff
@@ -74,7 +76,9 @@ static void gpio_irq_handler(void) {
 void cyw43_irq_init(void) {
     gpio_add_raw_irq_handler_with_order_priority(CYW43_PIN_WL_HOST_WAKE, gpio_irq_handler, CYW43_SHARED_IRQ_HANDLER_PRIORITY);
     irq_set_enabled(IO_IRQ_BANK0, true);
+    #if !defined(__riscv)
```

Comment:
Mostly out of curiosity, does the PICO 2W with Wi-Fi work in RISC-V mode?

Reply:
yes

Reply:
> we'd be able to switch the soft timer implementation back to the alarm pool

The "bug" is that an alarm handler taking <1us can cause future repeating alarms to not occur. This is fixed in develop by https://github.com/raspberrypi/pico-sdk/pull/2127. The proposal is to only replace the soft timer in mp_wfe_or_timeout with best_effort_wfe_or_timeout which handles repeated events properly and would allow the rp2350 to sleep. best_effort_wfe_or_timeout is not actually impacted by the issue addressed by https://github.com/raspberrypi/pico-sdk/pull/2127